src/xenia/base/threading_win.cc

```diff
@@ -315,8 +315,7 @@ class Win32Thread : public Win32Handle<Thread> {
   ~Win32Thread() = default;
 
   void set_name(std::string name) override {
-    AssertCallingThread();
-    xe::threading::set_name(name);
+    xe::threading::set_name(handle_, name);
```

Comment:
We need that assert - you can only set the thread name from the thread, not from any other thread. The code calling this needs to change to do that.